bin/tx-prover/src/error.rs

```diff
@@ -0,0 +1,23 @@
+use axum::http::uri::InvalidUri;
+use thiserror::Error;
+
+// TX PROVER ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum TxProverServiceError {
+    #[error("invalid uri")]
+    InvalidURI(#[source] InvalidUri),
+    #[error("failed to connect to worker")]
+    ConnectionFailed(#[source] tonic::transport::Error),
+    #[error("failed to create backend for worker")]
+    BackendCreationFailed(#[source] Box<pingora::Error>),
+    #[error("failed to setup pingora")]
+    PingoraConfigFailed(String),
```

Comment:
Shouldn't this either 1) include the `String` in the error message or 2) use `#[source]` on the `String` (which I don't recall whether it is even possible)?

Reply:
Yes, I've missed that. I added the String to the error message.